pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -73,16 +76,50 @@ func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDB) (bool, error
 }
 
 // getTLSConfigModification creates a modification function which enables TLS in the automation config.
-// The config is only updated after the certs and keys have been rolled out to all pods.
-// The agent needs these to be in place before the config is updated.
-// Once the config is updated, the agents will gradually enable TLS in accordance with: https://docs.mongodb.com/manual/tutorial/upgrade-cluster-to-ssl/
-func getTLSConfigModification(mdb mdbv1.MongoDB) automationconfig.Modification {
-	if !(mdb.Spec.Security.TLS.Enabled && hasRolledOutTLS(mdb)) {
-		return automationconfig.NOOP()
+// It will also ensure that the combined cert-key secret is created.
+func getTLSConfigModification(getUpdateCreator secret.GetUpdateCreator, mdb mdbv1.MongoDB) (automationconfig.Modification, error) {
+	if !mdb.Spec.Security.TLS.Enabled {
+		return automationconfig.NOOP(), nil
+	}
+
+	err := ensureTLSSecret(getUpdateCreator, mdb)
+	if err != nil {
+		return automationconfig.NOOP(), err
+	}
+
+	// The config is only updated after the certs and keys have been rolled out to all pods.
+	// The agent needs these to be in place before the config is updated.
+	// Once the config is updated, the agents will gradually enable TLS in accordance with: https://docs.mongodb.com/manual/tutorial/upgrade-cluster-to-ssl/
+	if hasRolledOutTLS(mdb) {
+		return tlsConfigModification(mdb), nil
+	} else {
+		return automationconfig.NOOP(), nil
+	}
+}
+
+// ensureTLSSecret will create or update the operator-managed Secret containing
+// the concatenated certificate and key from the user-provided Secret.
+func ensureTLSSecret(getUpdateCreator secret.GetUpdateCreator, mdb mdbv1.MongoDB) error {
+	data, err := secret.ReadStringData(getUpdateCreator, mdb.TLSSecretNamespacedName())
+	if err != nil {
+		return err
 	}
 
+	combined := fmt.Sprintf("%s%s", data[tlsSecretCertName], data[tlsSecretKeyName])
+
+	operatorSecret := secret.Builder().
+		SetName(mdb.TLSOperatorSecretNamespacedName().Name).
+		SetNamespace(mdb.TLSOperatorSecretNamespacedName().Namespace).
+		SetField(tlsOperatorSecretFileName, combined).
+		Build()
```

Comment:
Question: should we set the owner reference of the secret to point to the MDB resource? I don't see a reason to keep the secret around.

Reply:
it is a good idea to have the ownerreference set to something, so the resources belong to something, for maintenance or cleanup purposes.

Reply:
Good! Added now.